chemprop/cli/train.py

```diff
@@ -542,170 +531,220 @@ def main(args):
-        args.warmup_epochs,
-        args.init_lr,
-        args.max_lr,
-        args.final_lr,
-    )
-    logger.info(model)
-
-    monitor_mode = "min" if model.metrics[0].minimize else "max"
-    logger.debug(f"Evaluation metric: '{model.metrics[0].alias}', mode: '{monitor_mode}'")
-
-    tb_logger = TensorBoardLogger(args.output_dir, "tb_logs")
-    checkpointing = ModelCheckpoint(
-        args.output_dir / "chkpts",
-        "{epoch}-{val_loss:.2f}",
-        "val_loss",
-        mode=monitor_mode,
-        save_last=True,
-    )
-    early_stopping = EarlyStopping("val_loss", patience=5, mode=monitor_mode)
-
-    trainer = pl.Trainer(
-        logger=tb_logger,
-        enable_progress_bar=True,
-        accelerator="auto",
-        devices=args.n_gpu if torch.cuda.is_available() else 1,
-        max_epochs=args.epochs,
-        callbacks=[checkpointing, early_stopping],
-    )
-    trainer.fit(model, train_loader, val_loader)
-
-    if test_loader is not None:
-        if args.task_type == "regression":
-            model.predictor.register_buffer("loc", torch.tensor(scaler.mean_).view(-1, 1))
-            model.predictor.register_buffer("scale", torch.tensor(scaler.scale_).view(-1, 1))
-        results = trainer.test(model, test_loader)[0]
-        logger.info(f"Test results: {results}")
-
-    p_model = args.output_dir / "model.pt"
-    input_scalers = [] # TODO: we should add descriptor scalers here
-    output_scaler = scaler
-    save_model(p_model, model, input_scalers, output_scaler)
-    logger.info(f"Model saved to '{p_model}'")
+            train_dset = make_dataset(train_data[fold_idx], args.rxn_mode)
+            val_dset = make_dataset(val_data[fold_idx], args.rxn_mode)
+
+            if no_cv:
+                logger.info(f"Training model")
+            else:
+                logger.info(f"Training model for fold {fold_idx + 1}/{args.num_folds}")
+
+            mp_cls = BondMessagePassing if bond_messages else AtomMessagePassing
+            if multicomponent:
+                mp_blocks = [
+                    mp_cls(
+                        train_dset.datasets[i].featurizer.atom_fdim,
+                        train_dset.datasets[i].featurizer.bond_fdim,
+                        d_h=args.message_hidden_dim,
+                        bias=args.message_bias,
+                        depth=args.depth,
+                        undirected=args.undirected,
+                        dropout=args.dropout,
+                        activation=args.activation,
+                    )
+                    for i in range(n_components)
+                ]
+                if args.mpn_shared:
+                    mp_block = MulticomponentMessagePassing(
+                        mp_blocks[0], n_components, args.mpn_shared
+                    )
+                else:
+                    mp_block = MulticomponentMessagePassing(
+                        mp_blocks, n_components, args.mpn_shared
+                    )
```

Comment:
```
mp_block = MulticomponentMessagePassing(
    mp_blocks[0] if args.mpn_shared else mp_blocks, n_components, args.mpn_shared
)
```

Reply:
Is this `if/else` necessary? It seems like it's taken care of inside [`MulticomponentMessagePassing.__init__`](https://github.com/chemprop/chemprop/blob/327c31e932db8cea92e9d9a221c288eaeee911cb/chemprop/nn/message_passing/multi.py#L50):
```python
self.blocks = nn.ModuleList([blocks[0]] * self.n_components if shared else blocks)
```

Reply:
@davidegraff Good call! You're right. I will remove this `if/else`.